fvm/src/kernel/mod.rs

```diff
@@ -72,13 +73,16 @@ pub trait Kernel:
     /// - `method` is the method that has been invoked.
     /// - `value_received` is value received due to the current call.
     /// - `blocks` is the initial block registry (should already contain the parameters).
+    #[allow(clippy::too-many-arguments)]
     fn new(
         mgr: Self::CallManager,
         blocks: BlockRegistry,
         caller: ActorID,
         actor_id: ActorID,
         method: MethodNum,
         value_received: TokenAmount,
+        gas_premium: TokenAmount,
+        gas_limit: u64,
```

Comment:
See above. Can get the limit from the `call_manager.gas_tracker`, and we should add the premium there too.

Reply:
ack.

Reply:
done.